### PR TITLE
Fix list_modes

### LINKED ISSUE #2469: pygame.display.list_modes() return value contains duplicates in Pygame 2
```
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pygame
pygame 2.0.1 (SDL 2.0.14, Python 3.8.5)
Hello from the pygame community. https://www.pygame.org/contribute.html
>>> pygame.display.init()
>>> pygame.display.list_modes()
[(1920, 1080), (1920, 1080)]
```
Note that the two items in the return value tuple are identical.

Running the same code on the same machine in in Pygame 1.9.6 gives the expected results:
```
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pygame
pygame 1.9.6
Hello from the pygame community. https://www.pygame.org/contribute.html
>>> pygame.display.init()
>>> pygame.display.list_modes()
[(1920, 1080)]
```

I am running Ubuntu 20.04.1 on a Lenovo Legion Y520 laptop.

### PR DESCRIPTION
**Pull request by MightyJosip**
This should fix #2469. The problem is that SDL differentiate modes on multiple criteria, while PyGame does so only based on width/height.

Before this PR on my computer:
```
l = pygame.display.list_modes()
print(l)
print(len(l))
print(len(set(l)))
```
results with:
```
[(2715, 1527), (1920, 1080), (1680, 1050), (1600, 1024), (1600, 900), (1366, 768), (1360, 768), (1280, 1024), (1280, 1024), (1280, 960), (1280, 960), (1280, 800), (1280, 800), (1280, 768), (1280, 768), (1280, 720), (1280, 720), (1152, 864), (1024, 768), (1024, 768), (800, 600), (800, 600), (720, 576), (720, 576), (720, 480), (720, 480), (640, 480), (640, 480)]
28
18
```
And after this PR I get
```
[(2715, 1527), (1920, 1080), (1680, 1050), (1600, 1024), (1600, 900), (1366, 768), (1360, 768), (1280, 1024), (1280, 960), (1280, 800), (1280, 768), (1280, 720), (1152, 864), (1024, 768), (800, 600), (720, 576), (720, 480), (640, 480)]
18
18
```

### CODE REVIEW

**Review by Starbuck5 — APPROVED** [review]
Looks good.

This assumes that the SDL video mode variants of each resolution are sequential, which they seem to be.

I tested this on my mac and I didn't experience the original issue, and this PR doesn't change the output at all, which is good.

File: src_c/display.c [hunk 1]
```diff
@@ -1607,6 +1608,11 @@ pg_list_modes(PyObject *self, PyObject *args, PyObject *kwds)
             Py_DECREF(list);
             return NULL;
         }
+        if (last_width == mode.w && last_height == mode.h && last_width != -1) {
+            continue;
```
Comment (nthykier) [inline src_c/display.c:1612]: Leaks `size` here.  My recommendation is to move this bit up before we create the `size` tuple to side step the issue.

Alternatively, we need a `Py_DECREF(size);` before the `continue;`

**Review by nthykier — CHANGES_REQUESTED** [review]

File: src_c/display.c (hunk 1 above)
Comment (illume) [reply src_c/display.c:1612]: > move this bit up before we create the size tuple to side step the issue
Indeed, this would fix it.

Comment (MightyJosip) [reply src_c/display.c]: Thank you @nthykier . Can you check it now to see if it is ok

**Review by nthykier — APPROVED** [review]

**Comment by illume** [comment]
- The failure on SDL1 was unrelated to this PR.
- I reran the appveyor failure, since it seemed intermittent... and it passes.

**Review by illume — APPROVED** [review]
👍 thanks

**Pull request merged**